include/beast/websocket/stream.hpp

```diff
@@ -214,6 +214,17 @@ class stream : public detail::stream_base
         wr_opcode_ = o.value;
     }
 
+    /// Set the permessage-deflate extension options
+    void
+    set_option(permessage_deflate const& o);
+
+    /// Get the permessage-deflate extension options
+    void
+    get_option(permessage_deflate& o)
```

Comment:
return types cannot be co-variant, so each `get_option` would need a different name. I'm just following the asio pattern. Its not perfect yet.